Ghidra/Processors/V850E2M/data/languages/Instructions/Special.sinc

```diff
@@ -0,0 +1,260 @@
+{
+	FEPC = inst_next;
+	FEPSW = PSW;
+    $(FECC) = op1114 + 0x31;    # exception code 0x31..0x3F
+	FEIC = op1114 + 0x31;	    # exception code 0x31..0x3F
+	$(EP) = 1;
+	$(ID) = 1;
+	$(NP) = 1;
+	PC = 0x30;
+	goto [PC];
+}
+
+# HALT - 0000011111100000|0000000100100000
+:halt is op0015=0x7E0; op1631=0x120
+{
+	#I have no clue
+}
+
+# LDSR reg2, regID - rrrrr111111RRRRR|0000000000100000
+:ldsr R0004, SR1115 is op0510=0x3F & SR1115 & R0004; op1631=0x20
+{
+	SR1115 = R0004;
+}
+
+# NOP - 0000000000000000
+:nop is op0015=0x0
+{
+	PC = inst_next;
+}
+
+# PREPARE list12, imm5 - 0000011110iiiiiL|LLLLLLLLLLL00001
+:prepare PrepList, prep0105 is prep0615=0x1E & prep0105 & prep1620=0x1 & PrepList
+{
+	build PrepList;
+	sp = sp - (prep0105 << 2);
+}
+
+# PREPARE list12, imm5, sp - 0000011110iiiiiL|LLLLLLLLLLL00011
+:prepare PrepList, prep0105, sp is prep0615=0x1E & prep0105 & prep1620=0x3 & PrepList & sp
+{
+	build PrepList;
+	sp = sp - (prep0105 << 2);
+	ep = sp;
+}
+
+# PREPARE list12, imm5, imm16 (low) - 0000011110iiiiiL|LLLLLLLLLLL01011|iiiiiiiiiiiiiiii
+:prepare PrepList, prep0105, s3247 is prep0615=0x1E & prep0105 & prep1620=0xB & PrepList; s3247
+{
+	build PrepList;
+	sp = sp - (prep0105 << 2);
+	ep = s3247;
+}
+
+# PREPARE list12, imm5, imm16 (high) - 0000011110iiiiiL|LLLLLLLLLLL10011|iiiiiiiiiiiiiiii
+:prepare PrepList, prep0105, s3247 is prep0615=0x1E & prep0105 & prep1620=0x13 & PrepList; s3247
+{
+	build PrepList;
+	sp = sp - (prep0105 << 2);
+	ep = s3247 << 16;
+}
+
+# PREPARE list12, imm5, imm32 - 0000011110iiiiiL|LLLLLLLLLLL11011|iiiiiiiiiiiiiiii|iiiiiiiiiiiiiiii
+:prepare PrepList, prep0105, imm32 is prep0615=0x1E & prep0105 & prep1620=0x1B & PrepList; op3247; op4863
+[ imm32 = (op4863 << 16) | op3247; ]
+{
+	build PrepList;
+	sp = sp - (prep0105 << 2);
+	ep = imm32;
+}
+
+# RETI - 0000011111100000|0000000101000000
+:reti is op0515=0x3F; op1631=0x140
+{
+    if($(EP)!=1)
+        goto <false>;
+
+    PC = EIPC;
+    PSW = EIPSW;
+    goto <end>;
+
+    <false>
+    if($(NP)!=1)
+        goto <false2>;
+
+    PC = FEPC;
+    PSW = FEPSW;
+    goto <end>;
+
+    <false2>
+    PC = EIPC;
+    PSW = EIPSW;
+
+    <end>
+    return[PC];
+}
+
+# RIE - 0000000001000000
+:rie is op0015=0x40 
+{
+    FEPC = PC;
+	FEPSW = PSW;
+	$(NP) = 1;
+    $(EP) = 1;
+	$(ID) = 1;
+	PC = 0x30;
+	goto [PC];
+}
+
+# RIE imm5, imm4 - iiiii1111111IIII|0000000000000000
+:rie op1115, op0003 is op0410=0x7F & op1115 & op0003; op1631=0x0
+{
+    FEPC = PC;
+	FEPSW = PSW;
+	$(NP) = 1;
+    $(EP) = 1;
+	$(ID) = 1;
+	PC = 0x30;
+	goto [PC];
+}
+
+# STSR regID, reg2 - rrrrr111111RRRRR|0000000001000000
+:stsr SR0004, R1115 is op0510=0x3F & R1115 & SR0004; op1631=0x40
+{
+	R1115 = SR0004;
+}
+
+# SWITCH reg1 - 00000000010RRRRR
+:switch r0004 is op0515=0x2 & r0004
+{
+	local adr:4 = inst_next + (r0004 << 1);
+	PC = inst_next + (sext(*:2 adr) << 1);
+	goto [PC];
+}
+
+# SYNCE - 0000000000011101
+:synce is op0015=0x1D 
+{
+	#I don't know that either
```

Comment:
you might add a pcodeop  `synce` and just call it here

Reply:
I can't find the Operation `synce` in the P-code Table of SLEIGH

Reply:
You can define your own:  `define pcodeop synce;`

In the function you can then call it:
```
{
    synce(); // you can have args here if you want, like an immediate or register
}
```
And then if it comes across that instruction the decompiler will insert `synce()`.  Just a suggestion so you can at least see the instruction hitting instead of a NOP

just for example, you can see their toy processor:
https://github.com/NationalSecurityAgency/ghidra/blob/bdb52bd73737db82507a963db81b101084037d98/Ghidra/Processors/Toy/data/languages/toy_builder.sinc#L61

Reply:
Alright, thank you!